File: app/src/main/java/com/taipei/yanghaobo/kunu/ui/API/BookListedVMFactory.java
package com.taipei.yanghaobo.kunu.ui.API;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.taipei.yanghaobo.kunu.data.KunuRepository;

public class BookListedVMFactory extends ViewModelProvider.NewInstanceFactory {
    private final KunuRepository mKunuRepository;

    public BookListedVMFactory(KunuRepository kunuRepository) {
        this.mKunuRepository = kunuRepository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        //noinspection unchecked
        return (T) new BookListedViewModel(mKunuRepository);
    }
}

File: app/src/main/java/com/taipei/yanghaobo/kunu/ui/api/BookListedVMFactory.kt
package com.taipei.yanghaobo.kunu.ui.api

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

import com.taipei.yanghaobo.kunu.data.KunuRepository

class BookListedVMFactory(private val mKunuRepository: KunuRepository) :
  ViewModelProvider.NewInstanceFactory() {

  override fun <T : ViewModel> create(modelClass: Class<T>): T {

    return BookListedViewModel(mKunuRepository) as T
  }
}
